src/software-management/self-update.md

```diff
@@ -0,0 +1,101 @@
+     
+  * Consequences: 
+     * Self-update can fail and could left the system in an unstable state. The device might be disconnected from Cloud.
+     * Potential fail reasons (without claim of completeness):
+       * No rollback of already processes package of one self-update request, when some subsequent package of same request fails.
+         <br/>-> leads to partially executed update. 
+       * Power-fail during update.
+         <br/>-> leads to partially executed update
+         <br/>-> can corrupt the package currently about to be updated
+         <br/>-> can corrupt the filesystem
+       * Installation of incompatibel set of thin-edge package versions (e.g. SM Agent of one release, and Mapper of another release).
+         <br/>-> thin-edge function might not given, when signifcant change happened between both release
+       * Tool(s) required by thin-edge installation procedure missing on the device (e.g. 'grep').
+         <br/>-> according package installation will stop.
+         
+  * Measures for more robusness:
+    * There are plans for thin-edge packages to reduce risks for installation failures:
+      <br/>TODO: For both, add some reference (some ‚ticket, discussion, ...)
+      1. Providing one thin-edge core package that contains everything needed for self-update (assumedly "Mapper", "SM Agent" and "SM Plugin"). 
+          So self-update just needs that one-and-only package; all other thin-edge packages can be update via usual Software Management.
+      2. Avoiding use of external tools in package installation logic (move from Shell scripts to Rust binary).
+    * If the custom device provides some transaction-safe package manager and some powerfail-safe filesystem, the device owner can 
+      provide an according Software Management plugin, and can make use of that by processing a self-update on the according module type. 
+
+*Hint: Before executing self-update on devices in the field it shall be tested on a kind of golden device that equals the device(s) in the field in system-environment, configuration and thin-edge version.*
+
+
+
+
+# Execution Flow and involved Components
+
+![Sequence Diagram Update SW-list](images/self-update.drawio.svg)
+
+
+# Design Principles
+
+* Thin-edge updates are managed by a specific software component named `self-updater`.
+
+* The self-updater is an own executable that implements the plugin API's command `update-list` (nothing more from the plugin API).
+
+* The SM Agent shall execute the self-updater instead of the SM plugin, for all thin-edge packages of an update request.
+
+* The SM Agent uses a "tedge-package list" to determine whether a package is a thin-edge package or not.
+
+* That "tedge-package list" holds all package-names that shall be managed by the self-updater. It shall allow to use regular expressions for package names.
+     * Simple exmaple 1: `[ "tedge*" ]`
+     * More fine grained exmaple 2: `[ "tedge_*", "tedge_agent_*", "tedge_apt_plugin_*" ]`
```

Comment:
I feel the plugins could be left out of this list as they can be updated independently. I doubt if updating an `apt plugin` or `apama plugin` would need all the heavy lifting in the self-update workflow. IMO, `tedge-self-updater` only needs to handle the "machine critical components" which are `tedge-agent` and the cloud mappers (not even the mappers like collectd mapper).

Reply:
That list is just an example to show the technical idea, and the list is configurable.

But anyway, good point. We need to define the default content of that list. We have a concerned follow-up topic mentioned in that spec (see "Providing one thin-edge core package that contains everything needed for self-update" in section "Measures for more robusness"). I will create a ticket for that, and add your point there.